packages/hardhat-core/src/internal/hardhat-network/provider/transactions/ReadOnlyValidEIP2930Transaction.ts

```diff
@@ -0,0 +1,178 @@
+  }
+
+  public static fromValuesArray(
+    values: AccessListEIP2930ValuesArray,
+    opts?: TxOptions
+  ): never {
+    throw new InternalError(
+      "`fromRlpSerializedTx` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+
+  private readonly _sender: Address;
+
+  public constructor(sender: Address, data: AccessListEIP2930TxData = {}) {
+    const fakeCommon = new Common({
+      chain: "mainnet",
+    });
+
+    // this class should only be used with txs in a hardfork that
+    // supports EIP-2930
+    (fakeCommon as any).isActivatedEIP = (eip: number) => {
+      if (eip === 2930) {
+        return true;
+      }
+
+      throw new Error("Expected `isActivatedEIP` to only be called with 2930");
+    };
+
+    // this class should only be used with EIP-2930 txs,
+    // which (we assume) always have a defined `chainId` value
+    (fakeCommon as any).chainIdBN = () => {
+      if (data.chainId !== undefined) {
+        return new BN(data.chainId);
+      }
+
+      throw new Error("Expected txData to have a chainId");
+    };
+
+    super(data, { freeze: false, common: fakeCommon });
+
+    this._sender = sender;
+  }
+
+  public verifySignature(): boolean {
+    return true;
+  }
+
+  public getSenderAddress(): Address {
+    return this._sender;
+  }
+
+  public sign(): never {
+    throw new InternalError(
+      "`sign` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+
+  public getDataFee(): never {
+    throw new InternalError(
+      "`getDataFee` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+
+  public getBaseFee(): never {
+    throw new InternalError(
+      "`getBaseFee` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+
+  public getUpfrontCost(): never {
+    throw new InternalError(
+      "`getUpfrontCost` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+
+  public validate(stringError?: false): never;
+  public validate(stringError: true): never;
+  public validate(stringError: boolean = false): never {
+    throw new InternalError(
+      "`validate` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+
+  public toCreationAddress(): never {
+    throw new InternalError(
+      "`toCreationAddress` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+
+  public getSenderPublicKey(): never {
+    throw new InternalError(
+      "`getSenderPublicKey` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+
+  public getMessageToVerifySignature(): never {
+    throw new InternalError(
+      "`getMessageToVerifySignature` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+
+  public getMessageToSign(): never {
+    throw new InternalError(
+      "`getMessageToSign` is not implemented in ReadOnlyValidTransaction"
+    );
+  }
+}
+
+// Override private methods
+
+const ReadOnlyValidTransactionPrototype: any =
+  ReadOnlyValidEIP2930Transaction.prototype;
+
+ReadOnlyValidTransactionPrototype._validateTxV = function () {};
```

Comment:
This shouldn't make sense here. There's no v validation. I think there's no need for any of them.